proposals/0345-android-12-issues.md

```diff
@@ -0,0 +1,322 @@
+                }
+            }
+            //If this app is the designated app we can start our RouterService
+            //Otherwise allow the designated app to start their RouterService
+            //...
+        }
+    }
+}
+
+//...
+
+private static SdlDeviceListener getSdlDeviceListener(Context context, BluetoothDevice bluetoothDevice) {
+
+    synchronized (DEVICE_LISTENER_LOCK) {
+        if (sdlDeviceListener == null) {
+            sdlDeviceListener = new SdlDeviceListener(context, bluetoothDevice, new SdlDeviceListener.Callback() {
+                @Override
+                public boolean onTransportConnected(Context context, BluetoothDevice bluetoothDevice) {
+                    //...
+                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+                        if (!AndroidTools.areBtPermissionsGranted(context, routerService.getPackageName()) && sdlAppInfoList.size() > 1) {
+                            for (SdlAppInfo appInfo : sdlAppInfoList) {
+                                if (AndroidTools.areBtPermissionsGranted(context, appInfo.getRouterServiceComponentName().getPackageName())) {
+                                    routerService = appInfo.getRouterServiceComponentName();
+                                    break;
+                                }
+                            }
+                        }
+                    }
+                    //...
+                }
+            }
+        }
+    }
+}
+~~~
+
+We also need to consider interactions with applications supporting older versions of the `SdlRouterService`. If an application using SDL Android 4.11 or older receives an `ACL_CONNECTED` broadcast in the `SdlBroadcastReceiver`, it may try to start up the `SdlRouterService` of an application that is targeting Android 12. If the started `SdlRouterService` does not have Bluetooth Permissions it will not be able to use the `bluetoothTransport`. To solve this issue we can update the `initCheck` within the `SdlRouterService` to check the apps Bluetooth Permissions. If the permissions are not granted we can fail the `initCheck` and try to deploy the next `SdlRouterService`.
+
+If a phone connects to the head unit over USB we can still start the `SdlRouterService` but if the designated app does not have Bluetooth Permissions the `bluetoothTransport` will still not be able to start. This could cause confusion for the user if they expect the `SdlRouterService` to connect over USB and Bluetooth but the `SdlRouterService` will only connect over USB. The suggested solution is to present a notification to the user reminding them to enable Bluetooth Permissions. By clicking on the notification the user will be directed to the apps settings page to grant those permissions. Meanwhile the `SdlRouterService` will wait to initialize the `bluetoothTransport` and will continuously check the permission status. Once the permissions are granted the `bluetoothTransport` will be started.
+
+#### Library Changes
```

Comment:
This header is already present on line 112. I think it can be deleted here.

Reply:
removed